### PR TITLE
[DCJ-261][risk=no] Always delete bee

### PR DESCRIPTION
**Pull request by rushtong**
### Addresses
https://broadworkbench.atlassian.net/browse/DCJ-261

### Summary
Ensure that BEEs are deleted even after creation failure

----
Have you read [CONTRIBUTING.md](../CONTRIBUTING.md) lately? If not, do that first.

- Label PR with a Jira ticket number and include a link to the ticket
- Label PR with a security risk modifier [no, low, medium, high]
- PR describes scope of changes
- Get a minimum of one thumbs worth of review, preferably two if enough team members are available
- Get PO sign-off for all non-trivial UI or workflow changes
- Verify all tests go green
- Test this change deployed correctly and works on dev environment after deployment

### CODE REVIEW

**Review by fboulnois — APPROVED** [review]
👍

File: .github/workflows/smoke-tests.yaml [hunk 1]
```diff
@@ -45,6 +45,7 @@ jobs:
           path: 'target/surefire-reports'
       - name: Bee Destroy
         uses: broadinstitute/workflow-dispatch@v4
+        if: always()
```
Comment (okotsopoulos) [inline .github/workflows/smoke-tests.yaml:48]: I'm curious -- if Bee fails to create, what is there to destroy?

**Review by okotsopoulos — APPROVED** [review]

File: .github/workflows/smoke-tests.yaml (hunk 1 above)
Comment (rushtong) [reply .github/workflows/smoke-tests.yaml:48]: > I'm curious -- if Bee fails to create, what is there to destroy?

I don't understand the internals completely, but it's a Sherlock thing. A failed bee creation can still end up with a Sherlock entry. The delete job ensures that Sherlock is purged of the failed bee.

**Review by aarohinadkarni — APPROVED** [review]
👍

File: .github/workflows/smoke-tests.yaml (hunk 1 above)
Comment (okotsopoulos) [reply .github/workflows/smoke-tests.yaml:48]: Thanks for the explanation!

**Pull request merged**